server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala

```diff
@@ -533,6 +581,22 @@ class ImpStream(partition: Int,
           )
         )
 
+        val checkRecoveryState = builder.add(
+          Flow[BGMessage[Command]].map { cmd =>
+            if (isRecoveryMode || isRecoveryModePriority) {
+              if(cmd.offsets.forall(o => isRecoveryMode && o.topic == persistCommandsTopic && o.offset >= initialWriteHead)) {
```

Comment:
As you told me, a command can have priority and non priority offsets and the logic need to be changed